dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -58,8 +60,10 @@ private[dynamodb] final case class DynamoDBExecutorImpl private (dynamoDb: Dynam
     constructor match {
```

Comment:
I think you can get rid of `.provideLayer(Clock.live)` by adding a constructor param that gets injected during ZLayer construction using `ZLayer.fromService`

```scala
private[dynamodb] final case class DynamoDBExecutorImpl private (clock: Clock.Service, dynamoDb: DynamoDb.Services)
    extends DynamoDBExecutor {
```

and gets constructed

```scala
object DynamoDBExecutor {
  val live: URLayer[DynamoDb with Clock, Has[DynamoDBExecutor]] =
    ZLayer
      .fromServices[Clock.Service, DynamoDb.Service, DynamoDBExecutor]((clock, dynamo) =>
        DynamoDBExecutorImpl(clock, dynamo)
      )
```

Reply:
I updated this but am not sure about passing the clock to the effects that need it. I currently have a `<effect>.provideLayer(ZLayer.succeed(clock))`. Does that make sense?

Reply:
changes look good, I have some more suggestions below to leverage `ZEnv`

Reply:
Updated this to use `Has`